src/Compiler/Utilities/illib.fs

```diff
@@ -14,7 +14,10 @@ open System.Runtime.CompilerServices
 [<Class>]
 type InterruptibleLazy<'T> private (value, valueFactory: unit -> 'T) =
     let syncObj = obj ()
+
+    [<VolatileField>]
     let mutable valueFactory = valueFactory
```

Comment:
Yeah, that's not going to work in preview version, b/c attribute targets enforcement feature @edgarfgp implemented.

Reply:
@vzarytovskii Yeah we are relying on `isFunTy cenv.g overallExprTy` which considers a lambda expression to be a function.

So we can either:
-  Account for this in the check for the codegen
- We extend the `VolatileField` attribute targets to `[<AttributeUsage(AttributeTargets.Field ||| AttributeTargets.Method, AllowMultiple=false)>]`. But now the `Field` in the name attribute is wrong.

Happy to raise a PR to fix this once we agreed on the approach.